doc/devel/gitwash/matplotlib_for_dev.rst

```diff
@@ -0,0 +1,20 @@
+.. _matplotlib-for-dev:
+
+=================================================
+Installing matplotlib from source for development
+=================================================
+
+After obtaining a local copy of the matpotlib source code (:ref:`set-up-fork`),
+navigate to the matplotlib directory and run the following in the shell:
+
+::
+    
+    python setup.py develop
```

Comment:
`pip install -e` just calls `setup.py develop`, so it doesn't matter much in practice right now. But the general movement is that `setup.py` is being deprecated and should always be called through `pip`, because in the hopefully-not-too-distant future `pip` will know how to invoke non-distutils-based build systems.

As a side note I hesitate to recommend `setup.py develop` *or* `pip install -e` because they kinda inherently screw up your virtualenv (you end up with skew between what pip thinks is installed, what python files you actually have installed, and what compiled files you have installed), but these kinds of installs are really popular anyway so I guess this is a losing battle...